app/controllers/api/v1/users_controller.rb

```diff
@@ -11,6 +11,20 @@ def index
         render json: users, status: :ok
       end
 
+      def destroy_self
+        authorize current_user, :destroy_self?
+
+        result = Users::DestroySelf.call(user: current_user, password: confirmation_password)
+
+        if result.success?
+          render json: { message: "Account deleted successfully" }, status: :ok
+        else
+          render json: { error: "Unable to delete account. Error: #{result.error}" }, status: :unprocessable_entity
+        end
+      rescue ServiceActor::Failure => e
```

Comment:
Usually we avoid having rescues on controller methods.
I think in this scenario you should not rescue here.